index.js

```diff
@@ -1,244 +1,370 @@
-
-		var typeChar = matches[1];
-
-		if (matches[4]) {
-			this.instance = parseInt(matches[4].substring(1), 10);
-		} else if (typeChar == 'U') {
-			this.instance = SteamID.Instance.DESKTOP;
+	/**
+	 * List of possible types
+	 * @static
+	 * @returns {{CHAT: number, P2P_SUPER_SEEDER: number, GAMESERVER: number, CLAN: number, ANON_USER: number, MULTISEAT: number, ANON_GAMESERVER: number, PENDING: number, CONTENT_SERVER: number, INVALID: number, INDIVIDUAL: number}}
+	 */
+	static get Type() {
+		return {
+			INVALID: 0,
+			INDIVIDUAL: 1,
+			MULTISEAT: 2,
+			GAMESERVER: 3,
+			ANON_GAMESERVER: 4,
+			PENDING: 5,
+			CONTENT_SERVER: 6,
+			CLAN: 7,
+			CHAT: 8,
+			P2P_SUPER_SEEDER: 9,
+			ANON_USER: 10
+		};
+	}
+
+	/**
+	 * List of named instances
+	 * @static
+	 * @returns {{ALL: number, CONSOLE: number, WEB: number, DESKTOP: number}}
+	 */
+	static get Instance() {
+		return {
+			ALL: 0,
+			DESKTOP: 1,
+			CONSOLE: 2,
+			WEB: 4
+		};
+	}
+
+	/**
+	 * Mapping of SteamID types to their characters
+	 * @static
+	 * @returns {object}
+	 */
+	static get TypeChars() {
+		return {
+			[SteamID.Type.INVALID]: 'I',
+			[SteamID.Type.INDIVIDUAL]: 'U',
+			[SteamID.Type.MULTISEAT]: 'M',
+			[SteamID.Type.GAMESERVER]: 'G',
+			[SteamID.Type.ANON_GAMESERVER]: 'A',
+			[SteamID.Type.PENDING]: 'P',
+			[SteamID.Type.CONTENT_SERVER]: 'C',
+			[SteamID.Type.CLAN]: 'g',
+			[SteamID.Type.CHAT]: 'T',
+			[SteamID.Type.ANON_USER]: 'a'
+		};
+	}
+
+	/**
+	 * Mask to be used to get the AccountID out of a 64-bit SteamID
+	 * @static
+	 * @returns {number}
+	 */
+	static get AccountIDMask() { return 0xFFFFFFFF; }
+
+	/**
+	 * Mask to be used to get the instance out of the upper 32 bits of a 64-bit SteamID
+	 * @static
+	 * @returns {number}
+	 */
+	static get AccountInstanceMask() { return 0x000FFFFF; }
+
+	/**
+	 * Flags in SteamID instance for chat type IDs
+	 * @static
+	 * @returns {{Lobby: number, Clan: number, MMSLobby: number}}
+	 */
+	static get ChatInstanceFlags() {
+		return {
+			Clan: (SteamID.AccountInstanceMask + 1) >> 1,
+			Lobby: (SteamID.AccountInstanceMask + 1) >> 2,
+			MMSLobby: (SteamID.AccountInstanceMask + 1) >> 3
+		};
+	}
+
+	/**
+	 * Create a new SteamID object.
+	 * @param {string|BigInt} [input] - BigInt containing 64-bit SteamID, or string containing 64-bit SteamID/Steam2/Steam3 text formats. If omitted, creates a blank SteamID object.
+	 */
+	constructor(input) {
+		this.universe = SteamID.Universe.INVALID;
+		this.type = SteamID.Type.INVALID;
+		this.instance = SteamID.Instance.ALL;
+		this.accountid = 0;
+
+		if (!input) {
+			// Use the default invalid values
+			return;
 		}
 
-		if (typeChar == 'c') {
-			this.instance |= SteamID.ChatInstanceFlags.Clan;
-			this.type = SteamID.Type.CHAT;
-		} else if (typeChar == 'L') {
-			this.instance |= SteamID.ChatInstanceFlags.Lobby;
-			this.type = SteamID.Type.CHAT;
+		let matches;
+		if (typeof input == 'bigint' || (typeof input == 'string' && input.match(/^\d+$/))) {
```

Comment:
Isnan or isfinite instead of regex better maybe?

Reply:
Negative numbers would pass both of those.